### PR TITLE
api: secure api programmaticaly

### PR DESCRIPTION
**Pull request by ioannistsanaktsidis**
WARNING: To be checked first , especially the part with invenio-accounts.

### CODE REVIEW

**Comment by psaiz** [comment]
Wait, this is to merge directly in qa. @tiborsimko : could you please confirm that this is correct?

**Comment by tiborsimko** [comment]
> Wait, this is to merge directly in qa. @tiborsimko : could you please confirm that this is correct?

I think we can just inspect how the new Invenio situation handles this sort of security in "master" branch, and adapt it accordingly. It should not be necessary to integrate this concrete patch into "qa" because (i) Invenio had been quite updated in the meantime and (ii) we'll soon have "master" generally available again. Hence I thought checking only new "master" in this respect.

**Comment by psaiz** [comment]
I'm still missing what this thing is meant to improve. If I use the endpoint to `PUT`, `POST`, I get `405 Not Allowed`. The GET works, as expected:
```
 % curl -X POST https://opendata-dev.cern.ch/api/records/21570 -d '{"hello":"world"}'
<html>
<head><title>405 Not Allowed</title></head>
<body>
<center><h1>405 Not Allowed</h1></center>
<hr><center>nginx</center>
</body>
</html>
 % curl -X POST https://opendata-dev.cern.ch/api/records/1000 -d '{"hello":"world"}'
<html>
<head><title>405 Not Allowed</title></head>
<body>
<center><h1>405 Not Allowed</h1></center>
<hr><center>nginx</center>
</body>
</html>
 % curl -X GET https://opendata-dev.cern.ch/api/records/1000 -d '{"hello":"world"}' 
{"created":"2023-11-29T16:15:59.912622+00:00","id":"1000","links":{"bucket":"https://opendata-dev.cern.ch/api/files/34d0c8b5-607b-4921-9a04-12184530d03b","self":"https://opendata-dev.cern.ch/api/records/1000"}
```
Am I doing the right kind of queries?

File: setup.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
 # Get the version string. Cannot be done with import!
 g = {}
 with open(os.path.join('cernopendata', 'version.py'), 'rt') as fp:
-    exec(fp.read(), g)
+    exec (fp.read(), g)
```
Comment (zzacharo) [inline setup.py:37]: is this space correct?

File: cernopendata/config.py [hunk 2]
```diff
@@ -55,6 +55,7 @@
 #: Switch of automatic building.
 # ASSETS_AUTO_BUILD = True
 
+ACCOUNTS_REGISTER_BLUEPRINT = False
```
Comment (zzacharo) [inline cernopendata/config.py:58]: this is needed eventually as the default is None and it is checked here https://github.com/inveniosoftware/invenio-accounts/blob/04d5171d9e508835f4c54998e4f327443929b75d/invenio_accounts/ext.py#L147

Comment (psaiz) [reply cernopendata/config.py:58]: Thanks @zzacharo. Let me create a new PR only with this change then.

**Comment by psaiz** [comment]
Addressed  in https://github.com/cernopendata/opendata.cern.ch/commit/cb6f05ba375adee53d428c1afebf332cf61e7757

**Pull request closed without merging**